components/spotify/spotify.app.mjs

```diff
@@ -107,6 +95,7 @@ export default {
       type: "string",
       label: "Playlist ID",
       description: "Select an existing playlist with \"Structured Mode\" enabled, or reference a specific [`playlist_id`](https://developer.spotify.com/documentation/web-api/#spotify-uris-and-ids) with \"Structured Mode\" disabled (for example, `3cEYpjA9oz9GiPac4AsH4n`).",
+      withLabel: true,
```

Comment:
In testing, `this.playlistId` (and other "`withLabel`" props defined in the app file) was a string rather than a `{value,label}` object. The `label` was only passed through the prop when the `withLabel` property was overridden/added to the prop definition in the component.

For example:

```js
  // ...
  props: {
    spotify,
    playlistId: {
      propDefinition: [
        spotify,
        "playlistId",
      ],
    },
  },
  async run() {
    const { playlistId } = this;
    console.log(playlistId);
  },
```
Current:`6u4RXuAaERkPElnTkXzOqR `
Expected:`{ value: "6u4RXuAaERkPElnTkXzOqR", label: "My Playlist #1" }`

**Workaround**

In `add-items-to-playlist.mjs`:
```diff
  props: {
    spotify,
    playlistId: {
      propDefinition: [
        spotify,
        "playlistId",
      ],
+      withLabel: true,
    },
```

Reply:
In the [`new-track-in-playlist`](https://github.com/PipedreamHQ/pipedream/blob/c8f168de85813ceee84b32da46e53b2c41daff21/components/spotify/sources/new-track-in-playlist/new-track-in-playlist.mjs) source, [`this.playlistIds`](https://github.com/PipedreamHQ/pipedream/blob/c8f168de85813ceee84b32da46e53b2c41daff21/components/spotify/sources/new-track-in-playlist/new-track-in-playlist.mjs#L51) is expected to be an array rather than `{ value, label }` object.

If `withLabel` is changed to be inherited (when using [`propDefinition`](https://pipedream.com/docs/components/api/#general)), I think we'll need to either override the `withLabel` property of [`playlistIds`](https://github.com/PipedreamHQ/pipedream/blob/c8f168de85813ceee84b32da46e53b2c41daff21/components/spotify/sources/new-track-in-playlist/new-track-in-playlist.mjs#L14) (to `false`), or sanitize/parse `this.playlistIds`.

Reply:
This is good to know -- will fix in the runtime. Will update when done. Thanks!

Reply:
`withLabel` should be inherited from `propDefinition` now and `withLabel: false` override should work now. Thanks for noticing this @js07 !

Reply:
Inheriting `withLabel` from  `propDefinition` in actions works for me!

I might be doing something wrong, but [`playlistIds`](https://github.com/PipedreamHQ/pipedream/blob/c8f168de85813ceee84b32da46e53b2c41daff21/components/spotify/sources/new-track-in-playlist/new-track-in-playlist.mjs#L14) in the [`new-track-in-playlist`](https://github.com/PipedreamHQ/pipedream/blob/c8f168de85813ceee84b32da46e53b2c41daff21/components/spotify/sources/new-track-in-playlist/new-track-in-playlist.mjs) **source** did not seem to inherit `withLabel` from `propDefinition` (using `pd dev`).

I also got an error when trying to deploy/publish a source or action component with a prop containing `withLabel: false`.
```
error         | component create api call responded with status: 500, body: {"error":"unexpected error, please contact support@pipedream.com"}
```

Reply:
I saw an issue with setting `withLabel: false` that is now fixed. I'm not sure I see why there should be a difference between source and action. Were you overriding in that case?

Reply:
> ~~Were you overriding in that case?~~

~~I wasn't overriding `withLabel` in that case. It was `true` in the app file's prop definition, but `this.playlistIds`, in the source, did not have labels.~~

I retested the `new-track-in-playlist` source and inheriting `withLabel` worked for me. I think I was doing something wrong.

`withLabel: false` also worked for me. Thank you!